### ISSUE TITLE
Performance issues with React 18 only on iOS

### ISSUE THREAD

**Issue by ashok3** [wix/react-native-calendars#2138]
## Description

I'm trying to use the Agenda component in our app using React 18 and React Native 0.70, but I have noticed that the Calendar part is updating very slowly and even incorrectly when we scroll the Reservations part fast on iOS. 

To diagnose this, I tried the examples in the master branch and I noticed that it's working perfectly, but that you are using React 17 and React Native 0.68 for the examples.

Just to double check, I created a new React Native project using React 17 and React Native 0.68, and it worked like your example. However, I also created a new React Native project using React 18 and React Native 0.70 and the problem I describe above happens. 

All this happens only on iOS. It seems fine on Android. I've provided both projects files here for you to compare this and see the issue clearly.

## Expected Behavior

I expect it to work the same or even better in React 18 than React 17.

## Observed Behavior

I observed that the Calendar part of the Agenda component was updating the selected date very slowly when using React 18.

## Environment

"react": "18.1.0",
"react-native": "0.70.6",
"react-native-calendars": "^1.1293.0"

Testing on simulator: iPhone 14 Pro (iOS 16)

## Reproducible Demo

Demo using React 18:
[MyAppReact18.zip](https://github.com/wix/react-native-calendars/files/10363666/MyAppReact18.zip)

Demo using React 17:
[MyAppReact17.zip](https://github.com/wix/react-native-calendars/files/10363624/MyAppReact17.zip)

## Screenshots
React 17  ✅
https://user-images.githubusercontent.com/716309/211112127-78ac7d73-bed7-4b32-8758-ec1d0b0295f1.mp4

React 18 ❌
https://user-images.githubusercontent.com/716309/211112967-dc1c5a16-87da-4adf-a85c-b8d81bc7eba5.mp4

**Comment by muhamdfikrii** [comment]
facing same issue

**Comment by GideonAgboba** [comment]
facing same issue

**Comment by devsales** [comment]
I'm having a performance issue with the Calendar & CalendarList components. It takes a long time to render and re-render (see video below). I haven't tested it on a real device though, only on the simulator.

"react": "18.1.0",
"react-native": "0.70.5",
"react-native-calendars": "^1.1293.0",

https://user-images.githubusercontent.com/47062834/212470336-21f7631d-2f8f-4ed5-9b06-931b0d7a2e47.mov

**Comment by muhamdfikrii** [comment]
> I'm having a performance issue with the Calendar & CalendarList components. It takes a long time to render and re-render (see video below). I haven't tested it on a real device though, only on the simulator.
> 
> "react": "18.1.0", "react-native": "0.70.5", "react-native-calendars": "^1.1293.0",
> 
>  Calendar.List.bad.performance.mov

performance down after upgraded to RN 0.70, working fine on RN 0.69

**Comment by devsales** [comment]
I'm using expo and tried to downgrade the expo SDK to v. 46 in order to use RN 0.69. I didn't notice any difference...

**Comment by ofilipowicz** [comment]
It is a dealbreaker issue that prevents us from using this library

**Comment by costaDZ** [comment]
I'm experiencing slowness when changing between months and days 

React native version : **0.70.5**

**Comment by channeladam** [comment]
Is there any active maintainer of this repo?

**Comment by andrewjmathers** [comment]
Same problem in android, works a little bit better in release mode, but I'd expect it to be a bit faster

**Comment by nirre7** [comment]
We see the same issue on Android and on Ios with CalendarList.
"react": "18.0.0",
"react-native": "0.69.7",
"react-native-calendars": "^1.1293.0",

**Comment by keithhackbarth** [comment]
We are experiencing this as well (similar to the video posted above) but don't have data points to know if it is an issue specific to react-native-calendars vs. react-native: 

https://github.com/facebook/react-native/issues/36296
https://github.com/facebook/react-native/issues/35778
https://github.com/facebook/react-native/issues/35778

I'm guessing that everyone that has reported this is using react-native ^0.7 with the legacy architecture. Is anyone experiencing it with the [new architecture](https://reactnative.dev/docs/new-architecture-intro)? Trying to decide if it is worth switching over and curious if this is the fix as it seems to be mentioned in the linked threads.

**Comment by keithhackbarth** [comment]
[@michaelbbb](https://github.com/MichalBBBB) - You mentioned in the linked thread that you were able to make some changes to improve performance. Would it be possible to share those changes here - I'd like to learn from what you did!

**Comment by conoremclaughlin** [comment]
> We are experiencing this as well (similar to the video posted above) but don't have data points to know if it is an issue specific to react-native-calendars vs. react-native:
> 
> [facebook/react-native#36296](https://github.com/facebook/react-native/issues/36296) [facebook/react-native#35778](https://github.com/facebook/react-native/issues/35778) [facebook/react-native#35778](https://github.com/facebook/react-native/issues/35778)
> 
> I'm guessing that everyone that has reported this is using react-native ^0.7 with the legacy architecture. Is anyone experiencing it with the [new architecture](https://reactnative.dev/docs/new-architecture-intro)? Trying to decide if it is worth switching over and curious if this is the fix as it seems to be mentioned in the linked threads.

Likewise curious to see what luck people have had with the new architecture and if that addresses many of the current performance issues.

**Comment by danibonilha** [comment]
> I'm having a performance issue with the Calendar & CalendarList components. It takes a long time to render and re-render (see video below). I haven't tested it on a real device though, only on the simulator.
> 
> "react": "18.1.0", "react-native": "0.70.5", "react-native-calendars": "^1.1293.0",

@devsales have you tried to use `NewCalendarList` ? I was having the same perf issues with `CalendarList` but there's this new one, which is one year old but is not documented

**Comment by devsales** [comment]
@danibonilha no, I haven't tried `NewCalendarList`. However I did test `CalendarList` on a real iOS device (iPhone 12) and the performance was better. It's not perfect but it was usable. Have you tried `NewCalendarList`?

**Comment by Villar74** [comment]
I tried `NewCalendarList`, but it have same picking performance issues, also as slower initial loading on simulator, so switched back to `CalendarList`

**Comment by kostas64** [comment]
Im facing the same issue only on iOS. Render is very slow. Android is rendering very fast.

  "react": "18.2.0",
  "react-native": "0.71.9",
  "react-native-calendars": "^1.1300.0",

https://github.com/wix/react-native-calendars/assets/50916347/6d87e7f4-3104-4815-9d28-c50a5f953f06

**Comment by NovakArtsiom** [comment]
Do we have any updates about the issue? I tried `NewCalendarList` on a device and simulator, but it works worse than `CalendarList`. 

"react": "18.2.0",
"react-native": "0.72.4",
"react-native-calendars": "^1.1301.0",

**Comment by guyromb** [comment]
Same issue

**Comment by vargajacint** [comment]
Hey folks, any update? We facing similar performance issues. Can we help somehow to resolve this issue?

**Comment by AlexMubarakshin** [comment]
Same issue

**Comment by vladyslavNiemtsev** [comment]
Same issue

**Comment by STonkoshkur** [comment]
It seems that this issue is not caused by React 18 itself, but by the new Hermes engine on iOS, which has been [enabled by default](https://reactnative.dev/blog/2022/09/05/version-070#hermes-as-default-engine) since RN 0.70.

Due to various issues caused by Hermes on iOS, our team has decided to disable the Hermes engine for iOS, resulting in a significant increase in calendar performance.

"react": "18.2.0",
"react-native": "0.72.8",
"react-native-calendars": "^1.1303.0",

**Comment by NikitaGura** [comment]
Thanks for info @STonkoshkur. For me disabling hermes helped. 

Solution for today is just add in Podfile:
```
  use_react_native!(
    **:hermes_enabled => false**
  )
```

**Comment by athul173** [comment]
Could something be done to handle the slow rendering without disabling heremes? Like showing a loading screen instead of the date string?

**Comment by norbiu** [comment]
@athul173 Did you end up finding a solution without disabling Hermes?

**Comment by danibonilha** [comment]
I ended up switching libraries to this one -> https://github.com/MarceloPrado/flash-calendar 
the performance is so much better and customization too

**Comment by AntonLukovtsev** [comment]
the same
